discussion/ownership-and-mutable-aliases.tex

```diff
@@ -1,6 +1,103 @@
 % !TEX root = ../main.tex
 
-\section{Ownership Applied to Hardware}
+\section{Applying Ownership to Hardware}
 \label{sec:ownership_allied_to_hardware}
 
-One of the initial problems that we wanted to investigate in this project, was to see if we could apply Rust's ownership-semantics directly to hardware.
+The problem that arises with aliasing to mutable data \cite{web:problem_with_shared_mutability}, also known as shared mutable state, might not be obvious to everyone, but this is often the root to problems like race conditions and other write-after-write issues.
+One of the initial problems that we wanted to investigate during this project, was to see if we could apply {\rust}'s ownership-semantics directly to hardware.
+As was described in \autoref{sec:impl:oo}, ARM \glspl{mcu}, like the EFM32 that we have targeted in this project, often comes bundled with a wide range of memory-mapped hardware peripherals.
+Obscure problems can occur if these peripherals are accessed from \emph{different} parts of a program \emph{at the same time}, this section discusses how such problems can arise and possibly presents a solution in which {\rust} might be used to prevent the issue, maybe.
```

Comment:
elaborate `at the same time`.

`the \emph{at the same time} here, can arise when an interrupt occurs and the interrupted code was accessing the same peripheral as the interrupt handler` eller noe.